### PR TITLE
Java: Add XREAD Command

### PR DESCRIPTION
**Pull request by acarbonetto**
This ticket is dependent on https://github.com/aws/glide-for-redis/pull/1494

### CODE REVIEW

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 1]
```diff
@@ -548,6 +549,7 @@ private static Object[] streamCommands(BaseTransaction<?> transaction) {
             "0-2", // xadd(streamKey1, Map.of("field2", "value2"), ... .id("0-2").build());
             "0-3", // xadd(streamKey1, Map.of("field3", "value3"), ... .id("0-3").build());
             3L, // xlen(streamKey1)
+            Map.of(streamKey1, Map.of("0-3", Map.of("field3", "value3"))), // xread(Map.of(key9, "0-2"));
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:552]: I think
```suggestion
            Map.of(streamKey1, Map.of("0-3", new String[] {"field3", "value3"})), // xread(Map.of(key9, "0-2"));
```

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 2]
```diff
@@ -105,4 +106,60 @@ public interface StreamBaseCommands {
...
+    /**
+     * Reads entries from the given streams.
+     *
+     * @apiNote When in cluster mode, all keys in <code>keysAndIds</code> must map to the same hash
+     *     slot.
+     * @see <a href="https://redis.io/commands/xread/">redis.io</a> for details.
+     * @param keysAndIds A <code>Map</code> of keys and entry ids to read from. The <code>
+     *     Map</code> is composed of a stream's key and the id of the entry after which the stream
+     *     will be read.
+     * @return A <code>{@literal Map<String, Map<String, Map<String, String>>>}</code> with stream
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:119]:
```suggestion
     * @return A <code>{@literal Map<String, Map<String, String[]>>}</code> with stream
```

Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:119]: or
```suggestion
     * @return A <code>{@literal Map<String, Map<String, String[][]>>}</code> with stream
```
as per https://github.com/aws/glide-for-redis/pull/1494#issuecomment-2138450640
Please also describe that most inner String [2D] array is a map-like structure, which dislike of a `Map` can store duplicates.

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 3]
```diff
@@ -105,4 +106,60 @@ public interface StreamBaseCommands {
...
+     *     keys, to <code>Map</code> of stream-ids, to a <code>Map</code> of field-entries.
+     * @example
+     *     <pre>{@code
+     * Map<String, String> xreadKeys = Map.of("streamKey", "readId");
+     * Map<String, Map<String, Map<String, String>>> streamReadResponse = client.xread(xreadKeys).get();
+     * for (var keyEntry : streamReadResponse.entrySet()) {
+     *     for (var streamEntry : keyEntry.getValue().entrySet()) {
+     *         for (var fieldEntry : streamEntry.getValue().entrySet()) {
+     *             System.out.printf("Key: %s; stream id: %s; field: %s; value: %s\n", keyentry.getKey(), streamEntry.getKey(), fieldEntry.getKey(), fieldEntry.getValue());
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:128]: Update examples too.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -2664,6 +2666,38 @@ public T xdel(String key, String[] ids) {
...
     }
 
+    /**
+     * Reads entries from the given streams.
+     *
+     * @see <a href="https://redis.io/commands/xread/">redis.io</a> for details.
+     * @param keysAndIds An array of <code>Pair</code>s of keys and entry ids to read from. A <code>
+     *     pair</code> is composed of a stream's key and the id of the entry after which the stream
+     *     will be read.
+     * @return Command Response - A <code>{@literal Map<String, Map<String, Map<String, String>>>}
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2676]: same

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 5]
```diff
@@ -4025,6 +4030,82 @@ public void xdel_returns_success() {
...
 
+    @SneakyThrows
+    @Test
+    public void xread_multiple_keys() {
+        // setup
+        String keyOne = "one";
+        String streamIdOne = "id-one";
+        String keyTwo = "two";
+        String streamIdTwo = "id-two";
+        Map<String, Map<String, Map<String, String>>> completedResult = new LinkedHashMap<>();
```
Comment (Yury-Fridlyand) [inline java/client/src/test/java/glide/api/RedisClientTest.java:4041]: And tests...

**Comment by acarbonetto** [comment]
instead of merging/rebasing, I opted to raise: https://github.com/aws/glide-for-redis/pull/1524

**Pull request closed without merging**